api/api/utils/image_proxy/__init__.py

```diff
@@ -0,0 +1,116 @@
+import logging
+from typing import Literal
+from urllib.parse import urlparse
+
+from django.conf import settings
+from django.http import HttpResponse
+from rest_framework.exceptions import UnsupportedMediaType
+
+import django_redis
+import requests
+import sentry_sdk
+
+from api.utils.image_proxy.exception import UpstreamThumbnailException
+from api.utils.image_proxy.extension import get_image_extension
+from api.utils.image_proxy.photon import get_photon_request_params
+from api.utils.tallies import get_monthly_timestamp
+
+
+parent_logger = logging.getLogger(__name__)
+
+HEADERS = {
+    "User-Agent": settings.OUTBOUND_USER_AGENT_TEMPLATE.format(
+        purpose="ThumbnailGeneration"
+    )
+}
+
+PHOTON_TYPES = {"gif", "jpg", "jpeg", "png", "webp"}
+ORIGINAL_TYPES = {"svg"}
+
+PHOTON = "photon"
+ORIGINAL = "original"
+THUMBNAIL_STRATEGY = Literal["photon_proxy", "original"]
+
+
+def get_thumbnail_strategy(ext: str | None) -> THUMBNAIL_STRATEGY | None:
+    """
+    We use photon for image types that are supported by photon (PHOTON_TYPES),
+    for SVG images we use the original image.
+    """
+    if ext in PHOTON_TYPES:
+        return PHOTON
+    elif ext in ORIGINAL_TYPES:
+        return ORIGINAL
+    return None
+
+
+def get(
+    image_url: str,
+    media_identifier: str,
+    accept_header: str = "image/*",
+    is_full_size: bool = False,
+    is_compressed: bool = True,
+) -> HttpResponse:
+    """
+    Proxy an image through Photon if its filetype is supported, return the original
+    image if the filetype is SVG, and raise .
+    """
+    logger = parent_logger.getChild("get")
+    tallies = django_redis.get_redis_connection("tallies")
+    month = get_monthly_timestamp()
+
+    image_extension = get_image_extension(image_url, media_identifier)
+    thumbnail_strategy = get_thumbnail_strategy(image_extension)
+    if not thumbnail_strategy:
+        raise UnsupportedMediaType(image_extension)
+
+    headers = {"Accept": accept_header} | HEADERS
+
+    parsed_image_url = urlparse(image_url)
+    domain = parsed_image_url.netloc
+
+    if thumbnail_strategy == ORIGINAL:
+        upstream_url = image_url
+        params = {}
+    else:
+        upstream_url, params, headers = get_photon_request_params(
+            parsed_image_url, is_full_size, is_compressed, headers
+        )
+    try:
+        upstream_response = requests.get(
+            upstream_url,
+            timeout=15,
+            params=params,
+            headers=headers,
```

Comment:
I think this is the most confusing part of this implementation to me, just because it's hard to understand what the effect of the strategy is. That might only be because I have an expectation based on the existing implementation, though.

It would be slightly easier, I think, there were distinct functions for each strategy that called to `requests.get`, but definitely adding a break between the `if/else` and `try` block would go a long way, otherwise the `try` looks somehow part of the `if/else` :flushed: 
```suggestion
    if thumbnail_strategy == ORIGINAL:
        upstream_url = image_url
        params = {}
    else:
        upstream_url, params, headers = get_photon_request_params(
            parsed_image_url, is_full_size, is_compressed, headers
        )

    try:
        upstream_response = requests.get(
            upstream_url,
            timeout=15,
            params=params,
            headers=headers,
```

Reply:
I've extracted a function that gets request parameters based on the image extension. The strategy selection is implicit there.